File: MeloRenta/src/gestorAplicacion_contratos/Factura.java
package gestorAplicacion_contratos;
import java.io.Serializable;
import java.time.LocalDate;
import java.util.*;

public class Factura implements Serializable {
    private double id;
    private boolean estado;
    private Contrato contrato;
    
    //inicio serializacion     
    private static final long serialVersionUID = 1L;
    private static LinkedList<Factura> facturas = new LinkedList<Factura>();
       
    
    public static LinkedList<Factura> getFacturas() {
		return facturas;
	}

	public static void adicionarFactura(Factura Factura) {
		facturas.add(Factura);
	}
    

	public static void setFacturas(LinkedList<Factura> facturas) {
		Factura.facturas = facturas;
	}

	//fin serialización   

    public Factura(double id, boolean estado, Contrato contrato) {
        this.id = id;
        this.estado = estado;
        this.contrato = contrato;
    }

    //getter and setter

    public double getId() {
        return id;
    }

    public void setId(double id) {
        this.id = id;
    }

    public boolean isEstado() {
        return estado;
    }

    public void setEstado(boolean estado) {
        this.estado = estado;
    }

    public Contrato getContrato() {
        return contrato;
    }

    public void setContrato(Contrato contrato) {
        this.contrato = contrato;
    }

    //fin getter and setter


    

    public void calendario_pago(){
    	LocalDate finicio= this.contrato.getFecha_inicio();
    	LocalDate ffin= this.contrato.getFecha_fin();
    	ArrayList<LocalDate> fechas = new ArrayList<>();
    	while (true) {
			if (ffin.isAfter(finicio)) {
				fechas.add(finicio);
				finicio.plusDays(30);
			}else {
				break;
			}
		}
    }
    //Este metodo se encarga de crear una lista de los días de pago en el intervalo entre la fecha inicio y la fecha fin.
    

    
    public String toString() {
        return "Factura{" +
                "id=" + id +
                ", estado=" + estado +
                ", Contrato=" + contrato +
                '}';
    }
    //Este método se encarga de presentar toda la información del objeto factura
}

File: MeloRentaPython/gestorAplicacion/gestorAplicacion_contratos/Factura.py
import datetime as dt

class Factura():

    def __init__(self, id, estado, contrato):
        self._id = id
        self._estado = estado
        self._contrato = contrato

    # Getters and Setters

    def getId(self):
        return self._id

    def setId(self, id):
        self._id = id

    def isEstado(self):
        return self._estado

    def setEstado(self, estado):
        self._estado = estado
    
    def getContrato(self):
        return self._contrato

    def setContrato(self, contrato):
        self._contrato = contrato

    # Fin Getters and Setters

    def calendarioPago(self):  # Este metodo se encarga de crear una lista de los días de pago en el intervalo entre la fecha inicio y la fecha fin.
        fInicio = self.contrato.getFechaInicio()
        fFin = self.contrato.getFechaFin()
        fechas = []
        while (True):
            if (fFin > fInicio):
                fechas.append(fInicio)
                fInicio += dt.timedelta(days=30)
            else:
                break;
        
        return fechas;

    def __str__(self):
        if self.getContrato() is not None:
            return "Factura{" + "id = " + str(self.getId()) + ", estado = " + str(self.isEstado()) + ", Contrato = " + self.getContrato().mostrarContrato() + '}'